packages/utils/crates/utils-db/src/rusty.rs

```diff
@@ -678,7 +678,10 @@ pub mod wasm {
                 .map_err(|_| io::Error::from(io::ErrorKind::Other))?;
 
             if written >= 0 {
-                Ok(written as usize)
+                // we flush after every successful write to ensure consistency
+                fsync_sync(self.0)
+                    .map(|_| written as usize)
+                    .map_err(|_| io::Error::from(io::ErrorKind::Other))
```

Comment:
This has the potential to become a significant bottleneck.

Reply:
I agree it could be at scale, but currently not having a consistent database presents a bigger problem.